addons/binding/org.openhab.binding.sleepiq/src/main/java/org/openhab/binding/sleepiq/handler/SleepIQDualBedHandler.java

```diff
@@ -0,0 +1,146 @@
+import static org.openhab.binding.sleepiq.SleepIQBindingConstants.CHANNEL_RIGHT_ALERT_DETAILED_MESSAGE;
+import static org.openhab.binding.sleepiq.SleepIQBindingConstants.CHANNEL_RIGHT_ALERT_ID;
+import static org.openhab.binding.sleepiq.SleepIQBindingConstants.CHANNEL_RIGHT_IN_BED;
+import static org.openhab.binding.sleepiq.SleepIQBindingConstants.CHANNEL_RIGHT_LAST_LINK;
+import static org.openhab.binding.sleepiq.SleepIQBindingConstants.CHANNEL_RIGHT_PRESSURE;
+import static org.openhab.binding.sleepiq.SleepIQBindingConstants.CHANNEL_RIGHT_SLEEP_NUMBER;
+import static org.openhab.binding.sleepiq.SleepIQBindingConstants.THING_TYPE_DUAL_BED;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.BridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.sleepiq.config.SleepIQBedConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.syphr.sleepiq.api.SleepIQ;
+import org.syphr.sleepiq.api.model.BedSideStatus;
+import org.syphr.sleepiq.api.model.BedStatus;
+
+/**
+ * The {@link SleepIQDualBedHandler} is responsible for handling channel state updates from the cloud service.
+ *
+ * @author Gregory Moyer - Initial contribution
+ */
+public class SleepIQDualBedHandler extends BaseThingHandler implements BedStatusListener {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPE_UIDS = Collections.singleton(THING_TYPE_DUAL_BED);
+
+    private final Logger logger = LoggerFactory.getLogger(SleepIQDualBedHandler.class);
+
+    private volatile String bedId;
+
+    public SleepIQDualBedHandler(final Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+
+        logger.debug("Verifying SleepIQ cloud/bridge configuration");
+
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "No cloud service bridge has been configured");
+            return;
+        }
+
+        if (!ThingStatus.ONLINE.equals(bridge.getStatus())) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            return;
+        }
+
+        ThingHandler handler = bridge.getHandler();
+        if (!(handler instanceof SleepIQCloudHandler)) {
```

Comment:
This was more about hardening the code than having a specific use case in mind. I imagine this might be possible due to some misconfiguration or other unhappy path.